spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Recurrent.scala

```diff
@@ -243,6 +263,15 @@ class Recurrent[T : ClassTag](var batchNormParams: BatchNormParams[T] = null)
     extend(outputSize)
 
     /**
+     * for quantization, we need do forward first for allocating the weight memory
+     */
+    quantizeOptim()
```

Comment:
Don't do this - it's too hacky; you just need to make sure every QunatizedTensor is properly initialized when it is constructed.